### PR TITLE
Cloud Deploy and Automated `oss-vdb-test` Staging

### PR DESCRIPTION
**Pull request by michaelkedar**
- Two Cloud Deploy pipelines (`gke-workers` and `osv-api`) have been created in the `oss-vdb` project, with staging & production targets
  - The production targets are actually connected to production. **Please don't press the promote button yet.** It should work but I haven't properly checked it.
- Automatically create releases for these pipelines on push to master, which deploys to `oss-vdb-test`.
- Also automatically deploy App Engine to `oss-vdb-test` on push to master.
- Tidied up some old and outdated deployment documentation.

### CODE REVIEW

File: deployment/clouddeploy/osv-api/clouddeploy.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+apiVersion: deploy.cloud.google.com/v1
+kind: DeliveryPipeline
+metadata:
+  name: osv-api
+description: osv api pipeline
```
Comment (andrewpollock) [inline deployment/clouddeploy/osv-api/clouddeploy.yaml:5]: Nit: "OSV API pipeline"

File: deployment/clouddeploy/osv-api/clouddeploy.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    profiles: [ staging ]
+  - targetId: production
+    profiles: [ prod ]
+---
+
+apiVersion: deploy.cloud.google.com/v1
+kind: Target
+metadata:
+  name: staging
+description: oss-vdb-test api instance
```
Comment (andrewpollock) [inline deployment/clouddeploy/osv-api/clouddeploy.yaml:18]: Nit: "oss-vdb-test API instance"

File: deployment/clouddeploy/osv-api/clouddeploy.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  - RENDER
+  - DEPLOY
+  serviceAccount: deployment@oss-vdb.iam.gserviceaccount.com
+---
+
+apiVersion: deploy.cloud.google.com/v1
+kind: Target
+metadata:
+  name: production
+description: oss-vdb api instance
```
Comment (andrewpollock) [inline deployment/clouddeploy/osv-api/clouddeploy.yaml:32]: Nit: "oss-vdb API instance"

File: deployment/build-and-stage.yaml [hunk 4]
```diff
@@ -27,23 +27,23 @@ steps:
 - name: gcr.io/cloud-builders/gcloud
   entrypoint: bash
   args:
-    - "-c"
-    - |
-      read current_build _ <<< $(gcloud builds list --filter=tags="build-and-stage" --format="value(id,create_time)" --sort-by=create_time --ongoing)
+  - '-c'
+  - |
+    read current_build _ <<< $(gcloud builds list --filter=tags="build-and-stage" --format="value(id,create_time)" --sort-by=create_time --ongoing)
```
Comment (andrewpollock) [inline deployment/build-and-stage.yaml:32]: You could remove `create_time` from the format and then you wouldn't need to have the awkward looking underscore as part of the `read`, right?

File: deployment/build-and-stage.yaml [hunk 5]
```diff
@@ -27,23 +27,23 @@ steps:
...
+    if [[ -z "${current_build}" ]]; then
+      echo "Could not find any matching builds."
+      exit 1
+    fi
+    while [[ "${current_build}" != "${BUILD_ID}" ]]; do
+      echo "Waiting for build ${current_build}"
+      sleep 30
+      read current_build _ <<< $(gcloud builds list --filter=tags="build-and-stage" --format="value(id,create_time)" --sort-by=create_time  --ongoing)
```
Comment (andrewpollock) [inline deployment/build-and-stage.yaml:40]: Same here:

You could remove `create_time` from the format and then you wouldn't need to have the awkward looking underscore as part of the `read`, right?

**Review by andrewpollock — APPROVED** [review]

File: deployment/build-and-stage.yaml (hunk 4 above)
Comment (michaelkedar) [reply deployment/build-and-stage.yaml:32]: I had tried that, but it didn't seem return the results in the correct order. My guess was that the `--sort-by` flag needs the in the output in order for it to sort.

Comment (andrewpollock) [reply deployment/build-and-stage.yaml:32]: Really? Strange:

```
$ gcloud --project oss-vdb builds list --filter=tags="build-and-stage" --format="value(id,create_time)" --sort-by=create_time
a6f9014c-c44e-4c15-9db3-2078bbf9008d    2023-02-16T06:21:17.870948332Z
a37d3ca6-7d8f-41c4-8f70-d3bb1bce4ecd    2023-02-16T23:23:14.650251760Z
47c260a7-6013-4318-a7e6-98bd06dffff5    2023-02-16T23:27:19.943816952Z
1465d5df-3d0d-457b-8868-b064ff4b1145    2023-02-21T22:31:40.402139Z
fc854b17-04f3-452a-8c37-036871559d2e    2023-02-21T23:34:34.953340Z
18c08d02-42dd-4344-9727-d780482aa1a2    2023-02-22T03:40:17.788380Z
4289cbdc-7e84-4cb5-9e6e-2684e67ee1b4    2023-02-23T00:01:52.660899Z
7774ca68-ac28-484d-a388-5c1459893835    2023-02-23T00:55:19.186404Z
f47161bd-b62c-45f9-b8a4-8b6542899423    2023-02-23T01:06:22.864975Z
38a71182-f22c-404c-80c0-7ed5afd77ab6    2023-02-23T04:23:11.050005Z
```

```
$ gcloud --project oss-vdb builds list --filter=tags="build-and-stage" --format="value(id)" --sort-by=create_time
a6f9014c-c44e-4c15-9db3-2078bbf9008d
a37d3ca6-7d8f-41c4-8f70-d3bb1bce4ecd
47c260a7-6013-4318-a7e6-98bd06dffff5
1465d5df-3d0d-457b-8868-b064ff4b1145
fc854b17-04f3-452a-8c37-036871559d2e
18c08d02-42dd-4344-9727-d780482aa1a2
4289cbdc-7e84-4cb5-9e6e-2684e67ee1b4
7774ca68-ac28-484d-a388-5c1459893835
f47161bd-b62c-45f9-b8a4-8b6542899423
38a71182-f22c-404c-80c0-7ed5afd77ab6
```

Comment (michaelkedar) [reply deployment/build-and-stage.yaml:32]: Hm I've just double checked this a few times in Cloud Build and it seems to be working fine.
Not sure how I was getting a wrong result before. Will remove `create_time`.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**